lib/src/client.dart

```diff
@@ -76,32 +104,44 @@ class ClientCall<Q, R> implements Response {
   StreamController<R> _responses;
   StreamSubscription<GrpcMessage> _responseSubscription;
 
-  final Map<String, String> metadata;
+  final CallOptions options;
+
+  Future<Null> _callSetup;
 
-  ClientCall(this._channel, this._method, {this.metadata = const {}}) {
+  ClientCall(this._channel, this._method, {this.options}) {
     _responses = new StreamController(onListen: _onResponseListen);
-    _call().catchError((error) {
+    _callSetup = _initiateCall().catchError((error) {
       _responses.addError(
           new GrpcError(1703, 'Error connecting: ${error.toString()}'));
     });
   }
 
-  Future<Null> _call() async {
+  Future<Null> _initiateCall() async {
     final connection = await _channel.connect();
+    final timeout = options?.timeout ?? _channel.options?.timeout;
     // TODO(jakobr): Populate HTTP-specific headers in connection?
     final headers = <Header>[
       _methodPost,
       _schemeHttp,
       new Header.ascii(':path', _method.path),
       new Header.ascii(':authority', _channel.host),
+    ];
+    if (timeout != null) {
+      headers.add(new Header.ascii('grpc-timeout', '${timeout.inSeconds}S'));
```

Comment:
I'm not, but my brain's stopped working for today. You'll notice that I'm also adding an extra timeout of 5S just below this... Fix coming up.